provers/cairo/src/air.rs

```diff
@@ -487,9 +529,7 @@ fn add_pub_memory_in_public_input_section(
     let mut a_aux = addresses.to_owned();
     let mut v_aux = values.to_owned();
 
-    let output_range = public_input.memory_segments.get(&MemorySegment::Output);
-
-    let pub_addrs = get_pub_memory_addrs(output_range, public_input);
+    let pub_addrs: Vec<Felt252> = public_input.public_memory.clone().into_keys().collect();
     let mut pub_addrs_iter = pub_addrs.iter();
```

Comment:
This looks like you just want to iterate the keys. Why copy the whole public memory?